velox/exec/fuzzer/RowNumberFuzzerBase.cpp

```diff
@@ -0,0 +1,269 @@
+#include "velox/exec/fuzzer/FuzzerUtil.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/serializers/CompactRowSerializer.h"
+#include "velox/serializers/PrestoSerializer.h"
+#include "velox/serializers/UnsafeRowSerializer.h"
+
+DEFINE_int32(steps, 10, "Number of plans to generate and test.");
+
+DEFINE_int32(
+    duration_sec,
+    0,
+    "For how long it should run (in seconds). If zero, "
+    "it executes exactly --steps iterations and exits.");
+
+DEFINE_int32(
+    batch_size,
+    100,
+    "The number of elements on each generated vector.");
+
+DEFINE_int32(num_batches, 10, "The number of generated vectors.");
+
+DEFINE_double(
+    null_ratio,
+    0.1,
+    "Chance of adding a null value in a vector "
+    "(expressed as double from 0 to 1).");
+
+DEFINE_bool(enable_spill, true, "Whether to test plans with spilling enabled.");
+
+DEFINE_int32(
+    max_spill_level,
+    -1,
+    "Max spill level, -1 means random [0, 7], otherwise the actual level.");
+
+DEFINE_bool(
+    enable_oom_injection,
+    false,
+    "When enabled OOMs will randomly be triggered while executing query "
+    "plans. The goal of this mode is to ensure unexpected exceptions "
+    "aren't thrown and the process isn't killed in the process of cleaning "
+    "up after failures. Therefore, results are not compared when this is "
+    "enabled. Note that this option only works in debug builds.");
+
+namespace facebook::velox::exec {
+
+RowNumberFuzzerBase::RowNumberFuzzerBase(
+    size_t initialSeed,
+    std::unique_ptr<test::ReferenceQueryRunner> referenceQueryRunner)
+    : vectorFuzzer_{getFuzzerOptions(), pool_.get()},
+      referenceQueryRunner_{std::move(referenceQueryRunner)} {
+  setupReadWrite();
+  seed(initialSeed);
+}
+
+void RowNumberFuzzerBase::setupReadWrite() {
+  filesystems::registerLocalFileSystem();
+  dwrf::registerDwrfReaderFactory();
+
+  if (!isRegisteredNamedVectorSerde(VectorSerde::Kind::kPresto)) {
+    serializer::presto::PrestoVectorSerde::registerNamedVectorSerde();
+  }
+  if (!isRegisteredNamedVectorSerde(VectorSerde::Kind::kCompactRow)) {
+    serializer::CompactRowVectorSerde::registerNamedVectorSerde();
+  }
+  if (!isRegisteredNamedVectorSerde(VectorSerde::Kind::kUnsafeRow)) {
+    serializer::spark::UnsafeRowVectorSerde::registerNamedVectorSerde();
+  }
+
+  // Make sure not to run out of open file descriptors.
+  std::unordered_map<std::string, std::string> hiveConfig = {
+      {connector::hive::HiveConfig::kNumCacheFileHandles, "1000"}};
+  test::registerHiveConnector(hiveConfig);
+}
+
+void RowNumberFuzzerBase::logInput(const std::vector<RowVectorPtr>& input) {
+  if (VLOG_IS_ON(1)) {
+    // Flatten inputs.
+    const auto flatInput = test::flatten(input);
+    VLOG(1) << "Input: " << input[0]->toString();
+    for (const auto& v : flatInput) {
+      VLOG(1) << std::endl << v->toString(0, v->size());
+    }
+  }
+}
```

Comment:
nit: AggregationFuzzerBase also has a logVectors() method. We can move that to a common place to reuse the code.

Reply:
Check https://github.com/facebookincubator/velox/pull/12300. I'll update this PR post its merge.

Reply:
Updated with https://github.com/facebookincubator/velox/pull/12300